sale_order_product_recommendation_elaboration/wizards/sale_order_recommendation.py

```diff
@@ -38,21 +38,24 @@ class SaleOrderRecommendationLine(models.TransientModel):
     _name = "sale.order.recommendation.line"
     _inherit = ["sale.order.recommendation.line", "product.elaboration.mixin"]
 
-    def _prepare_elaborations_line_form(self, line_form):
+    def _prepare_elaborations_line_vals(self, vals):
```

Comment:
I think these extra prepare methods are not needed, as you can do (and override the same) in the main method.